src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -138,7 +138,17 @@ public void On(string methodName, Type[] parameterTypes, Func<object[], Task> ha
 
         private async Task<ReadableChannel<object>> StreamAsyncCore(string methodName, Type returnType, object[] args, CancellationToken cancellationToken)
         {
-            var irq = InvocationRequest.Stream(cancellationToken, returnType, GetNextId(), _loggerFactory, out var channel);
+            var irq = InvocationRequest.Stream(CancellationToken.None, returnType, GetNextId(), _loggerFactory, out var channel);
+            if (cancellationToken.CanBeCanceled)
+            {
+                cancellationToken.Register(async () =>
+                {
+                    if (!_connectionActive.IsCancellationRequested)
+                    {
+                        await SendInvocation(new CancelInvocationMessage(irq.InvocationId), irq);
```

Comment:
Rename SendInvocation to something else because now we send more than invocations.